iceoryx_posh/source/roudi/roudi_process.cpp

```diff
@@ -126,21 +134,154 @@ ProcessManager::ProcessManager(RouDiMemoryInterface& roudiMemoryInterface,
     m_memoryManagerOfCurrentProcess = m_segmentInfo.m_memoryManager;
 }
 
-void ProcessManager::killAllProcesses() noexcept
+void ProcessManager::killAllProcesses(const units::Duration processKillDelay) noexcept
 {
     std::lock_guard<std::mutex> g(m_mutex);
+    cxx::vector<bool, MAX_PROCESS_NUMBER> processStillRunning(m_processList.size(), true);
+    int i = 0;
+
+    auto yield = []() { std::this_thread::sleep_for(std::chrono::milliseconds(250)); };
 
     // send SIGTERM to all running applications
-    typename ProcessList_t::iterator it = m_processList.begin();
-    const typename ProcessList_t::iterator itEnd = m_processList.end();
+    for (auto& process : m_processList)
+    {
+        // if it was killed we need to check if it really has terminated, if we can't kill it, we consider it
+        // terminated
+        processStillRunning[i] = requestShutdownOfProcess(process, ShutdownPolicy::SIG_TERM, ShudownLog::FULL);
+        ++i;
+    }
 
-    for (; itEnd != it; ++it)
+    // wait for process to complete
+    bool allProcessFinished = false;
+    posix::Timer finalKillTimer(processKillDelay);
+
+    yield();
```

Comment:
Why are we waiting here?
I think we are using `yield()` only once so I would recommend writing here directly `std::this_thread::sleep_for(std::chrono::milliseconds(250));`

Reply:
Give processes some time to execute the terminate before checking on them. Code has been re-ordered.